packages/ckit/src/tx-builders/ChequeWithdrawBuilder.ts

```diff
@@ -0,0 +1,116 @@
+import { Address, utils, since, Cell } from '@ckb-lumos/base';
+import { SearchKey } from '@ckitjs/mercury-client';
+import { Amount, Builder, RawTransaction, Transaction, CellInput } from '@lay2/pw-core';
+import { CkbTypeScript, CkitProvider } from '..';
+import { AbstractPwSenderBuilder } from './pw/AbstractPwSenderBuilder';
+import { byteLenOfHexData, byteLenOfLockOnlyCell, mergeLockOnlyCells } from './builder-utils';
+import { Pw } from '../helpers/pw';
+import { BN } from '../helpers';
+import { bytes } from '@ckitjs/utils';
+import { minimalCellCapacity } from '@ckb-lumos/helpers';
+
+export interface ChequeWithdrawOptions {
+  sender: Address;
+  receiver: Address;
+  sudt?: CkbTypeScript;
+}
+
+export class ChequeWithdrawBuilder extends AbstractPwSenderBuilder {
+  constructor(private options: ChequeWithdrawOptions, provider: CkitProvider) {
+    super(provider);
+  }
+
+  async build(): Promise<Transaction> {
+    const { sender, receiver, sudt } = this.options;
+    const provider = this.provider;
+
+    const senderLock = provider.parseToScript(sender);
+    const receiverLock = provider.parseToScript(receiver);
+
+    const searchKey: SearchKey = {
+      script: provider.newScript(
+        'CHEQUE',
+        `0x${utils.computeScriptHash(receiverLock).slice(2, 42)}${utils.computeScriptHash(senderLock).slice(2, 42)}`,
+      ),
+      ...(sudt ? { filter: { script: sudt } } : {}),
+      script_type: 'lock',
+    };
+    const unwithdrawnCells = await provider.collectCells({ searchKey }, (cells) => cells.length <= 1000);
```

Comment:
maybe we should check if a cell is enable to withdraw, by checking if a cell has produced's block_number that is enough epochs away from the current block_number